ntoskrnl/rtl/misc.c

```diff
@@ -62,6 +62,121 @@ RtlGetVersion(IN OUT PRTL_OSVERSIONINFOW lpVersionInformation)
     return STATUS_SUCCESS;
 }
 
+/**
+ * @RtlGetNtProductType
+ *
+ * Retrieves the NT type product of the operating system. This is the kernel-mode variant
+ * of this function.
+ *
+ * @param[out]  ProductType
+ *      The NT type product enumeration value returned by the call.
+ *
+ * @return
+ *      The function returns TRUE when the call successfully returned the type product of the system.
+ *      It'll return FALSE on failure otherwise. In the latter case the function will return WinNT
+ *      as the default product type.
+ *
+ * @remarks
+ *      The call expects to be called at PASSIVE_LEVEL. This is the base implementation of the function
+ *      as there's stuff still left to be investigated. For instance the function uses ExAllocatePoolWithTag
+ *      with the tag as "iPtR". In addition to that the function checks if the product type is actually valid
+ *      by checking the "ProductTypeIsValid" member of _KUSER_SHARED_DATA structure. Currently we do not
```

Comment:
Would it be an idea to implement this code?
So that this function is 'complete' as soon as the path setting the ProductTypeIsValid to true is implemented?